### PR TITLE
[desktop] Add search feature for chat list

### LINKED ISSUE #6429: filter chats to quickly find a channel or contact
[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a user, I want to filter existing chats so that I can easily find a channel or contact chat.

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

After using Status every day it can be common to have 50+ chats containing both public channels and 1-1 chats. With so many chats it can be very hard to quickly return to an existing chat and involves lots of scrolling and careful scrutiny.

Let's give users a way to quickly search or filter chats to find existing channels or contacts.

#### Expected behavior
Search bar to filter chats

#### Actual behavior
Scrolling and studying the open chats for a long time to find a chat

### Additional Information

[comment]: # (Android or iOS.)
* Operating System: All

### PR DESCRIPTION
**Pull request by yenda**
Fixes #6429

### Summary:

Add search feature for chat list

#### Platforms (optional)
- macOS
- Linux

### Steps to test:
- Open an account that has a lot of chats
- On top of chat list type something in the search field
- Only chats that match the search should be displayed

status: ready

screenshot from 2018-10-19 16-29-34
screenshot from 2018-10-19 16-29-48

### CODE REVIEW

File: src/status_im/ui/screens/desktop/main/tabs/home/views.cljs [hunk 1]
```diff
@@ -65,17 +65,54 @@
...
+
 (views/defview chat-list-view []
-  (views/letsubs [home-items [:home-items]]
+  (views/letsubs [search-filter       [:search/filter]
+                  search-filter-ref   (atom nil)
+                  filtered-home-items [:search/filtered-home-items]]
     [react/view {:style styles/chat-list-view}
      [react/view {:style styles/chat-list-header}
-      [react/view {:style {:flex 1}}]
+      #_[react/view {:style {:flex 1}}]
```
Comment (janherich) [inline src/status_im/ui/screens/desktop/main/tabs/home/views.cljs]: Please remove commented out/debug code

File: src/status_im/ui/screens/desktop/main/tabs/home/views.cljs [hunk 2]
```diff
@@ -65,17 +65,54 @@
...
+                      :default-value search-filter
+                      :on-change (fn [e]
+                                   (let [native-event (.-nativeEvent e)
+                                         text         (.-text native-event)]
+                                     (re-frame/dispatch [:search/filter-changed text])))}]])
+
 (views/defview chat-list-view []
-  (views/letsubs [home-items [:home-items]]
+  (views/letsubs [search-filter       [:search/filter]
+                  search-filter-ref   (atom nil)
```
Comment (janherich) [inline src/status_im/ui/screens/desktop/main/tabs/home/views.cljs]: Why do we keep reference to the search filter react component ? I see it's passed to the `search-input` but it's not used there at all, if that's the case and I didn't miss something, please remove it.

**Review by janherich — CHANGES_REQUESTED** [review]

**Review by janherich — APPROVED** [review]

**Comment by chadyj** [comment]
@yenda Can you please add some screenshots for @EugeOrtiz @denis-sharypin @errorists?

Also, designers, how about ideas for this feature on mobile?

File: src/status_im/ui/screens/desktop/main/tabs/home/views.cljs [hunk 3]
```diff
@@ -65,17 +65,51 @@
   [react/touchable-highlight {:on-press #(re-frame/dispatch [:chat.ui/navigate-to-chat chat-id])}
    [chat-list-item-inner-view (assoc chat :chat-id chat-id)]])
 
+(defn tag-view [tag {:keys [on-press]}]
+  [react/touchable-highlight {:style {:border-radius 5
+                                      :margin 2
```
Comment (dmitryn) [inline src/status_im/ui/screens/desktop/main/tabs/home/views.cljs:70]: align map values

File: src/status_im/ui/screens/desktop/main/tabs/home/views.cljs [hunk 4]
```diff
@@ -65,17 +65,51 @@
...
+                                      :padding 4
+                                      :height 23
+                                      :background-color (if (= tag "clear filter")
+                                                          colors/red
+                                                          colors/blue)
+                                      :justify-content :center
+                                      :align-items :center
+                                      :align-content :center}
```
Comment (dmitryn) [inline src/status_im/ui/screens/desktop/main/tabs/home/views.cljs:78]: do we have agreement to move stylings to styles.cljs or not? just checking our styleguide

**Review by dmitryn — APPROVED** [review]

**Comment by chadyj** [comment]
Designers + @yenda, we can probably merge this and do a separate issue for design tweaks and mobile. We also need to consider message search and whether this is per-chat, or global.  WDYT?

**Review by vitvly — APPROVED** [review]

**Review by vitvly — APPROVED** [review]

**Comment by churik** [comment]
Tested on Mac OSx and Linux. Searched for 1-1 for users with default and custom usernames, public and group chats.
Mobile builds are not affected (tested on IOS).
Waiting for e2e tests to send it to `Merge` if no regression found.

**Pull request merged**